UnicornEngine/video/source/vulkan/Memory.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/vulkan/Memory.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+namespace vulkan
+{
+Memory::Memory() : m_memory(nullptr)
```

Comment:
What would happen if some user code copies it and calls free two or more times on the same memory?